### ISSUE TITLE
unstable sorting of files list in json files creates SCM noise

### ISSUE THREAD

**Issue by focusaurus** [tinganho/l10ns#99]
I am seeing my localization json files change just due to the `files` property not being deterministically sorted. For example:

```
     "files": [
-      "app/account/confirm-payment.jade",
       "app/account/billing.jade",
+      "app/account/confirm-payment.jade",
       "app/register/confirm-payment.jade"
     ],
```

Could this be sorted so it's stable in git unless a meaningful change happens?

**Comment by tinganho** [comment]
@focusaurus thanks for reporting. Now the files property is sorted before writing to storage. NPM published.

**Comment by focusaurus** [comment]
I'm still seeing issues in version 2.0.17. I can run `l10ns update` and see `files` key order changes. I can `git checkout` my locale json files to reset them, run `l10ns update` again and see a different set of changes. This is consistently reproducible.

In what may be an unrelated issue, sometimes `l10ns update` produces significant changes in my locales files, including losing values, when I would expect no change. I haven't been able to track this down yet.

And just FYI in terms of features, I have a bunch of data that does not live in source code primarily. It lives in a database  as values for HTML select elements. Thus before I can run `l10ns update` I have a script that pulls the values out of the database and generates a fake source code file with synthetic calls to `l("L_THIS_CAME_FROM_A_DB");`. I'd like to get out of these gymnastics at some point. 

Given the realities of a complex web app, I'm not sure the notion of "every key you need lives in source code files that l10ns knows how to parse" is a tenable design. I've got stuff in .js files, stuff in .jade templates, angularjs HTML/JS, values in databases, etc.

**Comment by tinganho** [comment]
@focusaurus I just tried to create a large set of files with many `l()` calls in them and run l10ns update on them and I can't reproduce this problem.

Can you give me a failing test? just create a public repo and let me pull it down?

**Comment by focusaurus** [comment]
OK I'll try to repro in a test repo tomorrow.

**Comment by tinganho** [comment]
@focusaurus thanks.

**Comment by focusaurus** [comment]
Here's a commit in a test repo demonstrating it.

https://github.com/focusaurus/test-l10ns/commit/5b5d7716e0203986b26378bbe0adcd176903033d

**Comment by focusaurus** [comment]
FYI this is on OSX. Maybe the case-insensitivity of the OSX filesystem is somehow weirdly involved? Just a hunch. In any case, this seems consistently reproducible for me.

I suspect this string subtraction is the problem:

https://github.com/tinganho/l10ns/commit/e8a821d09f24ad34041569e78c377e2ca8d96d12#diff-3693763d79803fa299e19d7b7fb53d62R77

```
> "a" - "b"
NaN
> "b" - "a"
NaN
> "a" > "b"
false
> "b" > "a"
true
```

**Comment by tinganho** [comment]
Thanks @focusaurus for you PR. NPM published now on l10ns@2.0.18.
